basics/barriers.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: content
+title: Barriers
+author: mark
+as_version: 3.0 canary 3
+cl_version: 1.1.0-beta1
+order: 5
+---
+### Barriers
+
+When we are creating Android layouts, sometimes we encounter situations where the layout can change depending on the localisation. Here is a really simple example:
+
+![Aligned Layout (English)](../assets/images/basics/barriers_alignment_en.png)
+
+Here we have three TextViews: `textView1` and `textView2` on the left; and `textView3` on the right. `textView3` has been constrained to the end of `textView1` and this works perfectly - it positions and sizes`textView3` exactly as we need it.
+ 
+ However, things become more complicated if we need to support multiple languages. If we add a German translation then we have a problem because in the English version the text in `textView1` is longer than that in `textView2`, whereas in German the text in `textView2` is longer than that in `textView1`:
+ 
+ ![Aligned Layout (German)](../assets/images/basics/barriers_alignment_de.png)
+
+The problem here is that `textView3` is still constrained to `textView1` and so `textView2` is now running in to `textView3`. This is seen most clearly in the Design view (the one with the white background).
+ 
+ The obvious solutions to this would be to either use `TableLayout`, or wrap `textView1` & `textView2` inside a vertical `LinearLayout` with `android:layout_width="wrap_content"` nd then constrain `textView3` to the end of this `LinearLayout`. But there is a better way: Barriers.
```

Comment:
In addition to this "and": Shouldn't this be "..., or _to_ wrap ..." right after the first comma? Adding the "to" sounds better to me. But, being a German, I might be wrong with that.